modules/representation_management/spec/requests/v0/pdf_generator_2122_spec.rb

```diff
@@ -59,29 +60,166 @@
       }
     end
 
-    context 'when submitting all required data' do
-      it 'responds with a created status' do
-        post(base_path, params:)
-        expect(response).to have_http_status(:created)
+    context 'when submitting valid data' do
+      context 'When submitting all fields' do
+        it 'responds with a ok status' do
+          post(base_path, params:)
+          expect(response).to have_http_status(:ok)
+        end
+
+        it 'responds with a PDF' do
+          post(base_path, params:)
+          expect(response.content_type).to eq('application/pdf')
+        end
       end
+    end
+
+    context 'when submitting valid data without optional fields' do
+      context 'When submitting all fields except claimant' do
```

Comment:
I find the nesting of context blocks a bit confusing:

```
context 'when submitting valid data without optional fields' do
  context 'When submitting all fields except claimant' do
  context 'When submitting none of the optional fields' do
```

I’m not sure the nesting adds much value here. Perhaps it would be clearer without the nested context blocks? For example:

```
context 'when submitting valid data without the optional claimant' do
end

context 'when submitting valid data without the optional claimant and without the optional veteran fields' do
end
```

This way, each scenario is clearly defined without additional nesting, making the tests easier to follow. What do you think?

Reply:
I reworked the wording and it's simplified now because I've reduced the request specs in favor of using model tests for the validations.